code/helpers/vector.dm

```diff
@@ -77,6 +82,14 @@ Inputs:
 		coord_x += inc_x
 		coord_y += inc_y
 
+		// The spinning x strikes the y in the z!
+		if(spin_counter != -1)
+			spin_counter++
+			if(spin_counter >= 8) spin_counter = 0
+			var/matrix/M = matrix()
+			M.Turn(90 * round(spin_counter/2))
```

Comment:
@Zuhayr hi I'm just lurking but uh:
```
var/matrix/M90 = matrix() //Cache these as statics if you want, I imagine 90,180,270 etc rot matrices might come in handy elsewhere + there's the benefit of not generating them on the fly.
M90.Turn(90)
var/matrix/M180 = matrix()
M180.Turn(180)
var/matrix/M270 = matrix()
M270.Turn(270)
var/matrix/M360 = matrix() //No turn, obviously.
animate(spining_thing, transform = M90, loop = -1, time = 2, /* extra settings */) //Loop = -1 is important
animate(transform = M180, time = 2, /* extra settings */)
animate(transform = M270, time = 2, /* extra settings */)
animate(transform = M360, time = 2, /* extra settings */)
```
That sets up an ```animate()``` "sequence" of transform -> M90, -> M180, -> M270, -> M360 -> M90...
As for making it not anti-alias, just have each step take no (or a very small, like the 2 above) time to complete.
Also just in case you don't know, missing out the ```spinning_thing``` from the 3 other calls is intentional that's how you do ```animate()``` sequences. The sequence will run until you animate() the object with a ref again (eg: ```animate(spinning_thing)``` would end the above sequence)

So I think that's what you want? I used something similar myself for ```orbit()```, it makes 36 (in default mode) matrices and swaps between them in an animate sequence (all steps after the first were done in a loop, which is how it can very in amount) to approximate a circle. Using animate like this, as @Loganbacca said above, made orbit entirely client side, which is perfect.

If you're interested in looking at the source for orbit, the matrices and animate stuff I did is here: https://github.com/tgstation/tgstation/blob/50a4dc19dca0477d538ad5da5829ac28c7165a11/code/__HELPERS/matrices.dm#L5 in ```SpinAnimation()```, Though it's AGPL code so it'd only be for curiosity/learning purposes.